pytest-embedded-qemu/pytest_embedded_qemu/qemu.py

```diff
@@ -15,7 +15,7 @@ class Qemu(DuplicateStdoutPopen):
     SOURCE = 'QEMU'
 
     QEMU_PROG_PATH = 'qemu-system-xtensa'
-    QEMU_DEFAULT_ARGS = '-nographic -no-reboot -machine esp32'
```

Comment:
We need to remove the `-no-reboot` option from `QEMU_DEFAULT_ARGS`  as the bootloader calls a software reset after flash encryption has been completed.

Reply:
Is it possible to override flags specifically for the case where flash encryption workflow is being tested? I'm afraid this is a breaking change, otherwise.

Reply:
@igrr

I made the same change [here](https://github.com/espressif/pytest-embedded/pull/179/files#diff-a6ffa4515a4b71193b9b1eaff9e1fd25a01965cfd4632d75d99696c91cf6aa65R20) to enable the qmp `system_reset` function.

IMHO this breaking change is not "breaking" for testing purpose... since the side effect is call system_reset and show the output again. for me it's more like a fix. WDYT?

Reply:
However if we had a test case which caused the emulated system to reset, thus finishing the test, will this test now just keep running? Do we currently assume that if QEMU process has exited, this means test failure?

Reply:
Thinking about this again, i agree that this change is okay.

What i was worried about is that without -no-reboot we could fail to detect a crash in a hypothetical application which crashes on first boot but succeeds on the 2nd one.

But now i understand that we won't correctly detect such application as failing even when testing on real hardware, so the QEMU case is not anyhow "worse".

Reply:
> Do we currently assume that if QEMU process has exited, this means test failure?

no we don't. the pexpect process (also could call it the dut process) is reading the qemu output from the qemu process, and doing the expect function calls based on that.